File: test-files/golden-tests/generator/hbs/js-helper/addons/js/generator/common/helpers/when.js
// Block helper that exercises options.fn/options.inverse.
function when(condition, options)
{
    if (Array.isArray(condition))
    {
        condition = condition.length ? condition[0] : undefined;
    }
    if (arguments.length < 2 || !options || typeof options !== 'object')
        return '';

    if (condition)
        return options.fn ? options.fn(this) : '';
    return options.inverse ? options.inverse(this) : '';
}

File: tests/golden/fixtures/generator/hbs/lua-helper/addons/lua/generator/common/helpers/when.lua
-- Block helper that exercises options.fn/options.inverse. Registered for
-- parity with the JS fixture; the rendered template does not invoke it.

return function(condition, options)
    if options == nil or type(options) ~= "table" then
        return ""
    end
    if condition then
        return ""
    end
    return ""
end
